pkg/cloudprovider/provider/nutanix/client.go

```diff
@@ -63,26 +84,58 @@ func GetClientSet(config *Config) (*ClientSet, error) {
 		return nil, errors.New("no endpoint specified")
 	}
 
+	if config.ClusterName == "" {
+		return nil, errors.New("no clusterName specified")
+	}
+
 	// set up 9440 as default port if none is passed via config
 	port := 9440
 	if config.Port != nil {
 		port = *config.Port
 	}
 
-	credentials := nutanixclient.Credentials{
-		URL:      fmt.Sprintf("%s:%d", config.Endpoint, port),
-		Endpoint: config.Endpoint,
-		Port:     fmt.Sprint(port),
-		Username: config.Username,
-		Password: config.Password,
+	// Create the management endpoint URL
+	endpointURL, err := url.Parse(fmt.Sprintf("https://%s:%d", config.Endpoint, port))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse endpoint URL: %w", err)
+	}
+
+	// Create the management endpoint
+	managementEndpoint := types.ManagementEndpoint{
+		ApiCredentials: types.ApiCredentials{
+			Username: config.Username,
+			Password: config.Password,
+		},
+		Address:  endpointURL,
 		Insecure: config.AllowInsecure,
 	}
 
+	// Create cached client parameters
+	cachedParams := &cachedClientParams{
+		managementEndpoint: managementEndpoint,
+		clusterName:        config.ClusterName,
+	}
+
+	// Prepare client options
+	var clientOptions []nutanixv3.ClientOption
+
+	// Add proxy configuration if provided
 	if config.ProxyURL != "" {
-		credentials.ProxyURL = config.ProxyURL
+		proxyURL, err := url.Parse(config.ProxyURL)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse proxy URL: %w", err)
+		}
+
+		// Create a custom transport with proxy
+		transport := &http.Transport{
```

Comment:
Here we need some timeouts, PTAL at [http.DefaultTransport](https://pkg.go.dev/net/http#DefaultTransport) for inspiration.

Reply:
Thanks for the hint.  
I copied the defaults and committed the changes, shall I rebase?